rfc/0009-audit-checks.md

```diff
@@ -35,67 +35,50 @@ As a Kubernetes developer, I want access to the audit checks results to allow me
-new field in the `ValidationRequest` in order to work with the audit checks and send back the
-result in the `ValidationResponse`. After the policies evaluations a [PolicyReport](https://github.com/kubernetes-sigs/wg-policy-prototypes/tree/master/policy-report)
-to store the results will be created.  Furthermore, it's necessary add configuration to allow
-users to define the periodicity of the audit check runs.
+The audit checks will require changes in `PolicyServer`, policies and the creation of a new component `Background Audit Scanner`.
 
 ## Policy server
 
-The audit checks will be executed in a dedicated the policy server running in parallel
-with others policy servers evaluating admission requests. This ensure that the audit
-checks will not slow down admission requests evaluations as well as remove the need
-of implementing a consensus algorithm to sync which policy server will run audit checks
-in a high availability setup.  In the "audit check" mode the policy server will
-trigger a job to run the policies validating the resources that the policies are
-configured to verify.
-
-The policy server needs a configuration to allow user to define policy serve mode
-(e.g. `audit`, `evaluation`) and the time between audit checks. Both can be a command
-line arguments as well as a environment variables.
-
-The policy server will not run multiple audit checks at the same time.
-This means that the next audit check should be scheduled just after the current one is finished.
-
-The policy server running in audit mode will be deployed using our Helm charts.
-
-This is a simple flowchart to show an overview how a policy server will work:
-
-```mermaid
-flowchart TD
-  id0(Policy Server start) --> id1{Is in audit mode?}
-  id1 -->|Yes|id7{Are there policies with audit enabled?}
-  id7 -->|Yes|id2(Set timer to trigger next audit check)
-  id2 -->id4(Wait audit check timer run off)
-  id4 -->id5(Run audit checks)
-  id5 -->id6(Report resources violating policies)
-  id6 -->id2
-  id1 -->|No|id3B(Evaluate admission requests)
-  id7 -->|No|id2b(Nothing to do)
-```
+`PolicyServer` needs to store successful evaluation from all policies, and also failed evaluations for 
+policies in monitor mode. Failed evaluations for policies in enforce mode will not be stored as resources won't be 
+created in the cluster. These evaluations will be stored in a `PolicyReport`. There is one `PolicyReport` per namespace and 
+one `ClusterPolicyReport` for cluster-wide resources, see this [RFC](https://github.com/kubewarden/rfc/pull/13/files) for more info.
+
+Many more calls will be made to `PolicyServer` for the background checks, therefore we need to implement a cache
+for policy evaluations. The result of each policy evaluation will be stored in the cache, it will contain the `resourceVersion`
+of the policy and the resource to see if any of them have changed. Before evaluating a new resource it will check in the 
+cache if evaluation was previously done. 
```

Comment:
what do we do with context-aware policies or those that make external calls?
Also, [raised in the PolicyReport RFC](https://github.com/kubewarden/rfc/pull/13#discussion_r965701469), it was said that not all requests will be audited (those performing UPDATEs for example can't be faked).

I think it would be a good idea to list specifically which kind of requests and which kind of policies are covered by the audit scan.

Reply:
I don't see a big problem with context-aware policies or those that make external calls. They would be evaluated with the context at this point in time, which give you an idea if this policy would reject a resource at this moment (it might change if the context changes).

Just CREATE requests will be audited, others operations should be ignored.

A operator can disable backgroundScan in any policy for cases like the [userinfo](https://github.com/kubewarden/rfc/pull/13#discussion_r965701469)

Reply:
I just think it's a good idea to document specifically which type of requests and which type of policies take play in the audit feature.
Resolving this thread, as I have raised that point in others.